site/src/helpers/planner.ts

```diff
@@ -184,6 +184,7 @@ function addMultiPlanToRoadmap(roadmap: SavedRoadmap | LegacyRoadmap): SavedRoad
       ],
       transfers: roadmap.transfers,
       timestamp: roadmap.timestamp,
+      currentPlanIndex: roadmap.currentPlanIndex,
```

Comment:
This took a while to understand that this addition is purely to preserve the data in roadmap, and is thus a bit confusing imo. (My first thought was: "why is this even in a function that only converts something to multiplan?" in fact, that line will probably never be called since `'planners' in roadmap` should always be true for plans with a current index)

To me, it would make more sense to _only override_ one property, i.e. return `{ ...roadmap, planners: [...] }` to make it more clear that you're returning a "modified copy of the original" rather than a "completely new `SavedRoadmap`"

Reply:
I think that reworking < 10 lines of logic when you add to it is a more reasonable decision than adding 1 line of logic that will need to be reworked to scale later. You are changing the functionality overall, so I am in favor of making that change properly.